Part1/ViewController.swift

```diff
@@ -0,0 +1,111 @@
+// I assume that most of the formatting issues come from the pdf conversion, therefore I upload a version which is
+// formatted by me
+
+// In my past projects when using UIKit, we preferred to build up ui programatically, without interface builder, since
+// it resuts faster code. I assume that in this task it is alright to use IB
+
+import UIKit
+
+func isIPhone() -> Bool {
+    return UIDevice.current.userInterfaceIdiom == .phone
+}
+
+class someVC:
+    UIViewController, UICollectionViewDelegate, UICollectionViewDelegateFlowLayout, UICollectionViewDataSource
+{
+    @IBOutlet weak var collectionViewTopConstraint: NSLayoutConstraint!
+    @IBOutlet var detailViewWidthConstraint: NSLayoutConstraint!
+    @IBOutlet var collectionView: UICollectionView!
+    @IBOutlet var detailView: UIView!
+    
+    var detailVC : UIViewController?
+    var dataArray : [Any]?
+    
+    override func viewWillAppear(_ animated: Bool) {
+        fetchData()
+        self.collectionView.dataSource = self
+        self.collectionView.delegate = self
+        self.collectionView.register(
+            UINib(nibName: "someCell", bundle: nil), forCellWithReuseIdentifier: "someCell")
+        
+        self.navigationItem.rightBarButtonItem = UIBarButtonItem.init(
+            title: NSLocalizedString("Done", comment: ""),
+            style: .plain, target: self,
+            action: #selector(dissmissController))
+    }
+    
+    @objc func dissmissController() {}
+    
+    func fetchData() {
+        let url = URL(string: "testreq")!
+
+        let task = URLSession.shared.dataTask(with: url) { [self](data, response, error) in
```

Comment:
Please use `weak self`, otherwise this code will result a memory leak.

```suggestion
        let task = URLSession.shared.dataTask(with: url) { [weak self] (data, response, error) in
             guard let strongSelf = self else { return }
```